server/models/procedures/stock.sql

```diff
@@ -643,4 +643,125 @@ CREATE PROCEDURE GetAMC(
     _initial_quantity AS quantity_at_beginning;
 END $$
 
+/* 
+ * RecomputeInventoryStockValue
+ * This procedure recompute the stock value for a given inventory
+ * and update value in the database, the value is computed
+ * in the enterprise currency
+ */
+DROP PROCEDURE IF EXISTS RecomputeInventoryStockValue$$
+CREATE PROCEDURE RecomputeInventoryStockValue(
+  IN _inventory_uuid BINARY(16),
+  IN _date DATE
+)
+BEGIN 
+  DECLARE v_cursor_all_movements_finished INTEGER DEFAULT 0;
+  
+  DECLARE v_quantity_in_stock INT(11) DEFAULT 0;
+  DECLARE v_wac DECIMAL(19, 4) DEFAULT 0;
+  DECLARE v_is_exit TINYINT(1) DEFAULT 0;
+
+  DECLARE v_line_quantity INT(11);
+  DECLARE v_line_unit_cost DECIMAL(19, 4);
+  DECLARE v_line_is_exit TINYINT(1);
+
+  DECLARE cursor_all_movements CURSOR FOR
+    SELECT sm.quantity, sm.unit_cost, sm.is_exit
+    FROM stock_movement AS sm
+    JOIN lot AS l ON l.uuid = sm.lot_uuid
+    JOIN inventory AS inv ON inv.uuid = l.inventory_uuid
+    JOIN depot d ON d.uuid = sm.depot_uuid
+    WHERE
+      d.is_cost_regulator = 1
+      AND inv.uuid = _inventory_uuid
+      AND DATE(sm.date) <= DATE(_date)
+    ORDER BY inv.text, DATE(sm.date), sm.created_at ASC;
+  
+  DECLARE CONTINUE HANDLER FOR NOT FOUND SET v_cursor_all_movements_finished = 1;
+
+  OPEN cursor_all_movements;
+
+  loop_cursor_all_movements : LOOP
+    FETCH cursor_all_movements INTO v_line_quantity, v_line_unit_cost, v_line_is_exit;
+
+    IF v_cursor_all_movements_finished = 1 THEN
+      LEAVE loop_cursor_all_movements;
+    END IF;
+
+    IF v_line_is_exit <> 0 THEN 
+      SET v_is_exit = -1;
+    ELSE 
+      SET v_is_exit = 1;
+    END IF;
+
+    IF v_line_is_exit = 0 AND v_quantity_in_stock > 0 THEN 
+      SET v_wac = ((v_quantity_in_stock * v_wac) + (v_line_quantity * v_line_unit_cost)) / (v_line_quantity + v_quantity_in_stock);
+    ELSEIF v_line_is_exit = 0 AND v_quantity_in_stock = 0 THEN
+      SET v_wac = (v_line_unit_cost * 1); /* we must use the exchange rate */
```

Comment:
It would be good to note that this only happens once, for the very first stock movement.